Doc review for Tekton Plugin (#265)
janus-idp/backstage-plugins: plugins/tekton/README.md

# Tekton plugin for Backstage

> The Tekton plugin helps with visualizing the workloads such as Deployment, Replicaset, and Pods powering any service on the Kubernetes cluster.

Reviewer: ```suggestion
The Tekton plugin helps with visualizing the `PipelineRun` resources available on the Kubernetes cluster.
```

Reviewer 2: What about TaskRuns should those be mentioned too?

Reviewer: As of today, we are visualizing only the pipelineRun, but we fetch the taskRuns of the associated pipeline to show the task status. Maybe frame it in a way so that users do not get the impression that they can visualize the task runs like how they do with Pipeline Runs.

Attaching the screenshot of the PipelineRun list page for ref:

<img width="1547" alt="PR" src="https://user-images.githubusercontent.com/22490998/233917164-6ba363fc-e97c-4c71-a5f7-ea53cca0759d.png">

Reviewer 3: Yeah needed task status as well to show the column task status bar viz

we do get taksStatus in the status of pipeline run status as well but we needed to fetch it separately as last I heard from tekton team is that task status in status of pipeline runs is getting depreciated so good to not rely on it.

Author: Done

## Prerequisites

> 1. Install and configure Kubernetes plugin by following the [installation](https://backstage.io/docs/features/kubernetes/installation) and [configuration](https://backstage.io/docs/features/kubernetes/configuration) guides.

Author: Done

> 2. Add the following `customResources` in [app-config.yaml](https://backstage.io/docs/features/kubernetes/configuration#configuring-kubernetes-clusters) file:

Author: not sure about the word section. However, changing the sentence to:

_Add the following `customResources` component in the [app-config.yaml]_

          apiVersion: 'v1beta1'
          plural: 'taskruns'
   ```

> 3. Kubernetes plugin is configured and able to connect to the cluster using a `ServiceAccount`.

Author: Done

> 4. The `ClusterRole` must be granted for custom resources (pipelineruns and taskruns) to `ServiceAccount` accessing the cluster. [ClusterRole](https://backstage.io/docs/features/kubernetes/configuration#role-based-access-control) is already granted if you have the Backstage Kubernetes Plugin configured.

Author: Changed the sentence to:

_The `ClusterRole` must be granted for custom resources (pipelineruns and taskruns) to `ServiceAccount` accessing the cluster. If you have the Backstage Kubernetes Plugin configured, the [ClusterRole](https://backstage.io/docs/features/kubernetes/configuration#role-based-access-control) is already granted. 
  You can use the following code to grant the `ClusterRole` for custom resources:_

   ```

> 5. To get the resources from a k8s cluster, add the following annotations to the entity's `catalog-info.yaml`:

Author: Done

   ```yaml
   annotations:
     backstage.io/kubernetes-id: <BACKSTAGE_ENTITY_NAME>
     backstage.io/kubernetes-namespace: <RESOURCE_NS>
   ```

>    The k8s plugin identifies if the provided entity contains k8s resources and from which namespace the plugin receives the resources based on the previous annotations.

Author: Done

> 6. A custom label selector can also be added which is used by Backstage to find the resources. The label selector takes precedence over the ID annotation.

Author: Done

   `'backstage.io/kubernetes-label-selector': 'app=my-app,component=front-end`

> 7. For k8s plugin to get the desired entity's k8s resources, the following label must be added to the resources(if label selector is used then the labels mentioned in that must be present on the resource):

Reviewer 4: ```suggestion
7. Add the following label to the resources, so that the Kubernetes plugin gets the Kubernetes resources from the requested entity (when using the label selector, the labels mentioned must be present on the resource):
```

Author: Changed the sentence to:

_7. You must add the following label to the resources, so that the Kubernetes plugin gets the Kubernetes resources from the requested entity:
   ```yaml
   'backstage.io/kubernetes-id': <BACKSTAGE_ENTITY_NAME>`
   ```
    ---
    **NOTE**
    
    When using the label selector, the mentioned labels must be present on the resource.
    
    ---_

## Development

> In [Backstage plugin terminology](https://backstage.io/docs/local-dev/cli-build-system#package-roles), the Tekton plugin is `frontend-plugin`. You can start a live dev session from the repository root using following command:

Reviewer 4: ```suggestion
In [Backstage plugin terminology](https://backstage.io/docs/local-dev/cli-build-system#package-roles), the Tekton plugin is a frontend plugin. You can start a live development session from the repository root using the following command:
```
